server/src/main/java/org/elasticsearch/transport/RemoteClusterConnection.java

```diff
@@ -640,7 +605,9 @@ public String executor() {
     }
 
     boolean assertNoRunningConnections() { // for testing only
-        assert connectHandler.running.availablePermits() == 1;
+        synchronized (connectHandler.mutex) {
```

Comment:
NIT: Probably no need for the synchronization here with the array blocking queue's `isEmpty` being thread safe anyway?